ntbs-service/Services/AuthorizationService.cs

```diff
@@ -10,63 +10,71 @@ namespace ntbs_service.Services
         {
-            this.userService = userService;
+            _userService = userService;
         }
 
         public async Task<bool> IsUserAuthorizedToManageAlert(ClaimsPrincipal user, Alert alert)
         {
-            var userTbServiceCodes = (await userService.GetTbServicesAsync(user)).Select(s => s.Code);
+            var userTbServiceCodes = (await _userService.GetTbServicesAsync(user)).Select(s => s.Code);
             return userTbServiceCodes.Contains(alert.TbServiceCode);
         }
 
         public async Task<bool> CanEdit(ClaimsPrincipal user, Notification notification)
         {
-            return (await FilterNotificationsByUserAsync(user, new Notification[] { notification })).Count() == 1;
+            if (_filter == null)
+            {
+                _filter = await GetUserPermissionsFilterAsync(user);
+            }
+
+            if (_filter.FilterByTBService)
+            {
+                return _filter.IncludedTBServiceCodes.Contains(notification.Episode.TBServiceCode);
+            }
+            else if (_filter.FilterByPHEC)
+            {
+                return _filter.IncludedPHECCodes.Contains(notification.Episode.TBService?.PHECCode)
+                    || _filter.IncludedPHECCodes.Contains(notification.PatientDetails.PostcodeLookup?.LocalAuthority?.LocalAuthorityToPHEC?.PHECCode);
+            }
+
+            return true;
         }
 
-        public async Task<IEnumerable<Notification>> FilterNotificationsByUserAsync(ClaimsPrincipal user, IEnumerable<Notification> notifications)
+        public async Task<IQueryable<Notification>> FilterNotificationsByUserAsync(ClaimsPrincipal user, IQueryable<Notification> notifications)
         {
-            if (filter == null)
+            if (_filter == null)
             {
-                filter = await GetUserPermissionsFilterAsync(user);
+                _filter = await GetUserPermissionsFilterAsync(user);
             }
     
-            if (filter.FilterByTBService)
+            if (_filter.FilterByTBService)
             {
-                notifications = notifications.Where(n => MatchesTBServiceCode(filter, n));
+                notifications = notifications.Where(n => _filter.IncludedTBServiceCodes.Contains(n.Episode.TBServiceCode));
             }
-            else if (filter.FilterByPHEC)
+            else if (_filter.FilterByPHEC)
             {
-                notifications = notifications.Where(n => MatchesPHECCode(filter, n));
+                notifications = notifications.Where(n => (n.Episode.TBService != null && _filter.IncludedPHECCodes.Contains(n.Episode.TBService.PHECCode))
+                                                        || (n.PatientDetails.PostcodeLookup != null && 
+                                                            n.PatientDetails.PostcodeLookup.LocalAuthority != null && 
+                                                            n.PatientDetails.PostcodeLookup.LocalAuthority.LocalAuthorityToPHEC != null && 

```

Comment:
Damn, that one is a bit of a mouthfull now too :P
I imagine putting things into variables might break the where clause though?
Perhpas splitting the first line across two lines, as well as reducing the indentation of the rest would help?